code/modules/projectiles/guns/ballistic/laser_gatling.dm

```diff
@@ -132,6 +139,24 @@
 
 /obj/item/gun/ballistic/minigun/process_fire(atom/target, mob/living/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0)
 	if(ammo_pack)
+		if(obj_flags & EMAGGED)
+			if(cooldown < world.time)
+				cooldown = world.time + 250
+				playsound(get_turf(src), 'sound/weapons/heavyminigunstart.ogg', 50, 0, 0)
+				slowdown = 5
+				bonus_spread = 50
+				sleep(15)
+				if(ammo_pack.overheat < ammo_pack.overheat_max)
+					playsound(get_turf(src), 'sound/weapons/heavyminigunshoot.ogg', 60, 0, 0)
+					for(var/i = 1 to burst_size)
+						addtimer(CALLBACK(src, .proc/process_burst, user, spread, i), fire_delay * (i - 1))
+					playsound(get_turf(src), 'sound/weapons/heavyminigunstop.ogg', 50, 0, 0)
+					slowdown = initial(slowdown)
+				else
+					to_chat(user, "The gun's heat sensor locked the trigger to prevent lens damage.")
+			else
+				to_chat(user, "You don't want your hands melt, let it cool down!.")
```

Comment:
missing a "To" here, although i to do not want my hands melt

Reply:
will get on that later